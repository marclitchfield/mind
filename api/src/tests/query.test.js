import { applySetup, verifyMutations } from './test-setup';
import gql from 'graphql-tag';

describe('Query tests', () => {
  applySetup();

  test('simple post (idea in space)', () => {
    return verifyMutations('Query tests', 'idea in space', {
      mutations: [
        ({space}) => gql`mutation { Idea { idea:post_in_space(input:{sourceId:"${space.id}", title: "test idea"}) { id } } }`
      ],
      query: ({idea}) => gql`query {
        Idea(id:"${idea.id}") { title spaces { title } }
      }`
    });
  });

  test('post entity with beforeMerge (event in space)', () => {
    return verifyMutations('Query tests', 'event in space', {
      mutations: [
        ({space}) => gql`mutation { Event { event:post_in_space(input:{sourceId:"${space.id}", title: "test event", datetime: "2011-12-12"}) { id } } }`
      ],
      query: ({event}) => gql`query {
        Event(id:"${event.id}") { title datetime spaces { title } }
      }`
    });
  });

  test('post instance (item in space)', () => {
    return verifyMutations('Query tests', 'item in space', {
      mutations: [
        ({space}) => gql`mutation { Concept { concept:post_in_space(input:{sourceId:"${space.id}", title: "test concept"}) { id } } }`,
        ({space, concept}) => gql`mutation { Item { item:post_in_space(input:{sourceId:"${space.id}", classId:"${concept.id}", title: "test item"}) { id } } }`
      ],
      query: ({item}) => gql`query {
        Item(id:"${item.id}") { title class { title itemInstances { title } } spaces { title } }
      }`
    });
  });

  test('post single cardinality (person at location)', () => {
    return verifyMutations('Query tests', 'person at location', {
      mutations: [
        ({space}) => gql`mutation { Location { location:post_in_space(input:{sourceId:"${space.id}", title: "test location"}) { id } } }`,
        ({location}) => gql`mutation { Person { person:post_at_location(input:{sourceId:"${location.id}", title: "test person"}) { id } } }`
      ],
      query: ({person}) => gql`query {
        Person(id:"${person.id}") { title location { title people { title } } spaces { title } }
      }`
    });
  });

  test('post single cardinality update (update person location)', () => {
    return verifyMutations('Query tests', 'update person location', {
      mutations: [
        ({space}) => gql`mutation { Location { location1:post_in_space(input:{sourceId:"${space.id}", title: "test location 1"}) { id } } }`,
        ({space}) => gql`mutation { Location { location2:post_in_space(input:{sourceId:"${space.id}", title: "test location 2"}) { id } } }`,
        ({location1}) => gql`mutation { Person { person:post_at_location(input:{sourceId:"${location1.id}", title: "test person"}) { id } } }`,
        ({person, location2}) => gql`mutation { Person { person:post_at_location(input:{id:"${person.id}", sourceId:"${location2.id}", title: "test person"}) { id } } }`
      ],
      query: ({person}) => gql`query {
        Person(id:"${person.id}") { title location { title people { title } } spaces { title } }
      }`
    });
  });

  test('post single cardinality instance (item at location)', () => {
    return verifyMutations('Query tests', 'item at location', {
      mutations: [
        ({space}) => gql`mutation { Concept { concept:post_in_space(input:{sourceId:"${space.id}", title: "test concept"}) { id } } }`,
        ({space}) => gql`mutation { Location { location:post_in_space(input:{sourceId:"${space.id}", title: "test location"}) { id } } }`,
        ({concept, location}) => gql`mutation { Item { item:post_at_location(input:{sourceId:"${location.id}", classId:"${concept.id}", title: "test person"}) { id } } }`
      ],
      query: ({item}) => gql`query {
        Item(id:"${item.id}") {
          class { title itemInstances { title } }
          location { title items { title } }
        }
      }`
    });
  });

  test('post properties (root concept in space)', () => {

  });

  test('post removal (idea in space)', () => {

  });
});