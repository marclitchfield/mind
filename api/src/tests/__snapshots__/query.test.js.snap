// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query tests post entity with beforeMerge (event in space) 1`] = `
Object {
  "Event": Object {
    "__typename": "Event",
    "datetime": "2011-12-12 00:00:00",
    "spaces": Array [
      Object {
        "__typename": "Space",
        "title": "event in space",
      },
    ],
    "title": "test event",
  },
}
`;

exports[`Query tests post instance (item in space) 1`] = `
Object {
  "Item": Object {
    "__typename": "Item",
    "class": Object {
      "__typename": "Concept",
      "itemInstances": Array [
        Object {
          "__typename": "Item",
          "title": "test item",
        },
      ],
      "title": "test concept",
    },
    "spaces": Array [
      Object {
        "__typename": "Space",
        "title": "item in space",
      },
    ],
    "title": "test item",
  },
}
`;

exports[`Query tests post properties (root concept in space) 1`] = `
Object {
  "Space": Object {
    "__typename": "Space",
    "concepts": Array [
      Object {
        "__typename": "Concept",
        "spaces": Array [
          Object {
            "__typename": "Space",
            "title": "root concept in space",
          },
        ],
        "title": "sub concept",
      },
      Object {
        "__typename": "Concept",
        "spaces": Array [
          Object {
            "__typename": "Space",
            "title": "root concept in space",
          },
        ],
        "title": "root concept",
      },
    ],
    "rootConcepts": Array [
      Object {
        "__typename": "Concept",
        "spaces": Array [
          Object {
            "__typename": "Space",
            "title": "root concept in space",
          },
        ],
        "title": "root concept",
      },
    ],
  },
}
`;

exports[`Query tests post single cardinality (person at location) 1`] = `
Object {
  "Person": Object {
    "__typename": "Person",
    "location": Object {
      "__typename": "Location",
      "people": Array [
        Object {
          "__typename": "Person",
          "title": "test person",
        },
      ],
      "title": "test location",
    },
    "spaces": Array [
      Object {
        "__typename": "Space",
        "title": "person at location",
      },
    ],
    "title": "test person",
  },
}
`;

exports[`Query tests post single cardinality instance (item at location) 1`] = `
Object {
  "Item": Object {
    "__typename": "Item",
    "class": Object {
      "__typename": "Concept",
      "itemInstances": Array [
        Object {
          "__typename": "Item",
          "title": "test person",
        },
      ],
      "title": "test concept",
    },
    "location": Object {
      "__typename": "Location",
      "items": Array [],
      "title": "test location",
    },
  },
}
`;

exports[`Query tests post single cardinality update (update person location) 1`] = `
Object {
  "Person": Object {
    "__typename": "Person",
    "location": Object {
      "__typename": "Location",
      "people": Array [
        Object {
          "__typename": "Person",
          "title": "test person",
        },
      ],
      "title": "test location 2",
    },
    "spaces": Array [
      Object {
        "__typename": "Space",
        "title": "update person location",
      },
    ],
    "title": "test person",
  },
}
`;

exports[`Query tests simple post (idea in space) 1`] = `
Object {
  "Idea": Object {
    "__typename": "Idea",
    "spaces": Array [
      Object {
        "__typename": "Space",
        "title": "idea in space",
      },
    ],
    "title": "test idea",
  },
}
`;
